### DISCUSSION TITLE
Companies / Sites using `@neshca/cache-handler`

### DISCUSSION THREAD

**Discussion by better-salmon** [caching-tools/next-shared-cache#487] [Show and tell]
Hello everyone,

I'm excited to witness the community's growth and eager to understand how various companies and sites utilize the `@neshca/cache-handler`. Whether you're a startup, a large corporation, or anything in between, your insights are invaluable to us. Please share your experiences and contribute to our collective knowledge!

Please share your experiences in this thread by posting:
1. A link to your website or repository.
2. Briefly describe how you use the `@neshca/cache-handler`.

Showcase Opportunity:
If you'd like to be featured on the Showcase page of the `@neshca/cache-handler` documentation site and README, please include:
- Your company's logo.
- A short review of your experience with the tool or the results you've achieved.

I am looking forward to seeing your contributions!

Thank you!

**Comment by erwanriou** [comment]
Hey here Archsplace founder (small start-up).

1. A link to your website or repository.

www.archsplace.com 

2. Briefly describe how you use the @neshca/cache-handler.

Well we started using it quite recently (approximately 2 week ago) and so far we are quite happy with it. But i have to say that the update from `1.0.8` to `1.3.1` and the new way to implement it make it much more resilient on the Redis side.

![Screenshot from 2024-04-25 18-12-33](https://github.com/caching-tools/next-shared-cache/assets/30130006/fa316446-6797-4bb1-89d6-39100d01a018)
the moment in the log we deployed the new version :)

So thanks for the good work!

**Reply by better-salmon** [reply]
Thank you for using it and for the reply. Could you please describe that new implementation method?

**Reply by erwanriou** [reply]
The new implementation method make sure to use the conditional `client?.isReady` as you suggested in your example and define a `defaultStaleAge` (better than infinity...)

```js
  if (client?.isReady && process.env.REDIS_AVAILABLE) {
    handler = await createRedisHandler({
      client,
      keyPrefix: `ARCHSPLACE_MAIN_${process.env.GIT_HASH}` ,
      timeoutMs: 5000
    })
  } else {
    handler = createLruHandler()
    console.warn("Falling back to LRU handler because Redis client is not available.")
  }
  
   return {
    handlers: [handler],
    ttl: {
      defaultStaleAge: 86400,
      estimateExpireAge: staleAge => staleAge * 2
    }
  }
```

also we switched over a new image of redis that is more fit for production: `redis/redis-stack-server:latest`. Since our nextjs deployment turn around 5 replicas down to 3 sometime, this solution avoid a lot of loss in cached data.
We use the app router with the cache from the `fetch` api.
